File: usvm-python/usvm-python-main/src/main/java/org/usvm/language/VirtualPythonObject.java
package org.usvm.language;

import org.usvm.machine.symbolicobjects.InterpretedInputSymbolicPythonObject;

public class VirtualPythonObject {
    public int interpretedObjRef;
    public VirtualPythonObject(InterpretedInputSymbolicPythonObject interpretedObj) {
        this.interpretedObjRef = interpretedObj.getAddress().getAddress();
    }
    public VirtualPythonObject(int interpretedObjRef) {
        this.interpretedObjRef = interpretedObjRef;
    }
}

File: usvm-python/usvm-python-main/src/main/kotlin/org/usvm/machine/interpreters/concrete/utils/VirtualPythonObject.kt
package org.usvm.machine.interpreters.concrete.utils

private const val MAX_NEEDED_MASK_BYTE_NUMBER: Int = 12
private const val ALL_SLOTS_BYTE: Int = 0b11111111

class VirtualPythonObject(
    @JvmField
    val interpretedObjRef: Int,
    val slotMask: ByteArray = List(MAX_NEEDED_MASK_BYTE_NUMBER) { ALL_SLOTS_BYTE.toByte() }.toByteArray(),
)
